src/util.rs

```diff
@@ -1,24 +1,16 @@
-use enums::*;
+use crate::enums::*;
 use libc::{c_char, c_uint};
-use raw;
+use log::warn;
 use std::fmt;
 use std::ffi::{CStr, CString};
 
-pub(crate) fn ptr_to_str(ptr: *const c_char) -> Option<&'static str> {
-    let slice : Option<&CStr> = unsafe {
-        if ptr.is_null() {
-            return None
-        }
-        Some(CStr::from_ptr(ptr))
-    };
-
-    match slice {
-        None => None,
-        Some(s) => {
-            let buf : &[u8] = s.to_bytes();
-            Some(std::str::from_utf8(buf).unwrap())
-        }
+pub(crate) unsafe fn ptr_to_str(ptr: *const c_char) -> Option<&'static str> {
```

Comment:
This function should definitely be unsafe - it assumes the pointer passed to it ends with '\0', and if not, it could page fault.

Reply:
Correct.